testament/testament.nim

```diff
@@ -897,28 +1093,65 @@ func extraOptions(run: TestRun): string =
   else:
     ""
 
-proc testSpecHelper(r: var TResults, run: var TestRun) =
+proc exeFile(
+    testRun: TestRun, specFilePath: string, rootDir: string): string =
+  ## Get name of the executable file for the test run
+  let
+    target = testRun.target
+    isJsTarget = target == targetJs
+    (dirPart, specName, _) = splitFile(specFilePath)
+    matrixEntry = testRun.matrixEntry
+    exeName =
+      (if matrixEntry == noMatrixEntry:
+        @[specName, target.cmd]
+      else:
+        @[specName, $matrixEntry, target.cmd]).join("_")
+    exeExt =
+      if isJsTarget:
+        "js"
+      else:
+        ExeExt
+
```

Comment:
This is one or block, extra spaces without rhyme or reason just make the code harder to read as it breaks flow, it's one continuous series of let definitions.